diff --git a/node_modules/expo-random/build/Random.js b/node_modules/expo-random/build/Random.js
index c25d62c..abfb2e0 100644
--- a/node_modules/expo-random/build/Random.js
+++ b/node_modules/expo-random/build/Random.js
@@ -19,6 +19,14 @@ function assertByteCount(value, methodName) {
 export function getRandomBytes(byteCount) {
     assertByteCount(byteCount, 'getRandomBytes');
     const validByteCount = Math.floor(byteCount);
+
+    if (__DEV__) {
+        const array = new Uint32Array(validByteCount);
+        for (let i = 0; i < validByteCount; i++) {
+            array[i] = Math.floor(Math.random() * 256);
+        }
+        return array;
+    }
     if (ExpoRandom.getRandomBytes) {
         return ExpoRandom.getRandomBytes(validByteCount);
     }
