diff --git a/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapTileProvider.java b/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapTileProvider.java
index 2b1a646..b940065 100644
--- a/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapTileProvider.java
+++ b/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapTileProvider.java
@@ -213,7 +213,9 @@ public class MapTileProvider implements TileProvider {
 			} catch (Exception e) {
 			  e.printStackTrace();
 			}
-		} else if (image == null && !this.offlineMode) {
+		}
+
+		if (image == null && !this.offlineMode) {
 			Log.d("urlTile", "Normal fetch");
 			image = fetchTile(x, y, zoom);
 			if (image == null) {
